### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by abdozayan12**
Screenshot 2023-09-06 113131

### CODE REVIEW

File: store_data.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+def store_data(app)
+  store_people(app)
+  store_books(app)
+  store_rentals(app)
+end
+
+def store_people(app)
+  people = []
+  storage = Set.new
```
Comment (danmainah) [inline store_data.rb:11]: - [ ] @abdozayan12 and @hmunish Great work adding a way to store data on app exit, however am getting the error below whenever I choose option 7, Kindly consider ensuring all options work and data is stored even when a JSON file is deleted
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @abdozayan12 and @hmunish , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Esnured on app startup all data from files is loaded.
✔️ Added functions to store data after the user exits the app.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: store_data.rb (hunk 1 above)
Comment (abdozayan12) [reply store_data.rb:11]: @danmainah 
Screenshot 2023-09-06 120802
When choosing option 7 it works fine

Comment (AbelG101) [inline store_data.rb:11]: - [ ] Hey guys @abdozayan12 @hmunish, It seems like I'm getting the same error as well I don't know how it's working on your machine but you can add `require 'set'` to properly import the set library and make the Set class available for use in your code. I've tried this and it's working fine on my end.
  image

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @abdozayan12 & @hmunish 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Correct use of Git Flow ✔
- CLI app works as expected ✔

 ## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  Implemented required changes. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**